test/common/quic/envoy_quic_client_session_test.cc

```diff
@@ -568,6 +568,7 @@ TEST_P(EnvoyQuicClientSessionTest, EcnReporting) {
   dispatcher_->run(Event::Dispatcher::RunType::NonBlock);
   const quic::QuicConnectionStats& stats = quic_connection_->GetStats();
   EXPECT_EQ(stats.num_ecn_marks_received.ect1, 1);
+  delete packet;
```

Comment:
nit: instead of explicitly calling delete here, can we make `packet` a `std::unique_ptr` instead up on line 549, perhaps via `absl::WrapUnique`?